rero_ils/modules/operation_logs/extensions.py

```diff
@@ -61,25 +63,42 @@ def _build_operation_log(self, record, operation):
             'user_name': 'system'  # default value, could be override
         }
         if current_librarian:
-            oplg.update({
+            oplg |= {
                 'user_name': current_librarian.formatted_name,
                 'user': {
                     'type': 'ptrn',
                     'value': current_librarian.pid
                 },
                 'organisation': {
                     'type': 'org',
-                    'value': current_librarian.organisation_pid
+                    'value': current_librarian.organisation_pid,
                 },
                 'library': {
                     'type': 'lib',
                     'value': current_librarian.library_pid
-                }
-            })
+                },
+            }
+            if current_library := flask_request.args.get('current_library'):
+                if current_library in \
+                        current_librarian.manageable_library_pids and (
+                            library := Library.get_record_by_pid(
+                                current_library)):
+                    oplg |= {
+                        'organisation': {
+                            'type': 'org',
+                            'value': extracted_data_from_ref(
+                                library.get('organisation'))
+                        },
+                        'library': {
+                            'type': 'lib',
+                            'value': library.get('pid')
+                        }
+                    }
```

Comment:
you should avoid to load `library` to get organisation using `current_librarian.organisation_pid`.
```python
if (lib_pid := flask_request.args.get('current_library')) and lib_pid in current_librarian.manageable_library_pids:
    oplg |= {
          'organisation': {
              'type': 'org',
              'value': current_librarian.organisation_pid
          },
          'library': {
              'type': 'lib',
              'value': lib.pid
          }
      }
```

Reply:
I did this to make sure I had the right organization for the library. It was for the future if a librarian can be admin in several organizations.